src/uct/ib/base/ib_md.c

```diff
@@ -469,40 +495,34 @@ static ucs_status_t uct_ib_md_reg_mr(uct_ib_md_t *md, void *address,
                                      int silent, uct_ib_mem_t *memh,
                                      uct_ib_mr_type_t mr_type)
 {
-    ucs_log_level_t level = silent ? UCS_LOG_LEVEL_DEBUG : UCS_LOG_LEVEL_ERROR;
     ucs_status_t status;
 
     if (length >= md->config.min_mt_reg) {
         UCS_PROFILE_CODE("reg ksm") {
             status = md->ops->reg_multithreaded(md, address, length,
-                                                access_flags, memh, mr_type);
+                                                access_flags, memh, mr_type,
+                                                silent);
         }
 
         if (status != UCS_ERR_UNSUPPORTED) {
             if (status == UCS_OK) {
                 memh->flags |= UCT_IB_MEM_MULTITHREADED;
-            } else {
-                uct_ib_md_print_mem_reg_err_msg(level, address, length,
-                                                access_flags);
             }
 
             return status;
         } /* if unsuported - fallback to regular registration */
     }
 
-    status = md->ops->reg_key(md, address, length, access_flags, memh, mr_type);
-    if (status != UCS_OK) {
-        uct_ib_md_print_mem_reg_err_msg(level, address, length, access_flags);
-        return status;
-    }
-
-    return UCS_OK;
+    return md->ops->reg_key(md, address, length, access_flags, memh, mr_type,
+                            silent);
 }
 
 ucs_status_t uct_ib_reg_mr(struct ibv_pd *pd, void *addr, size_t length,
-                           uint64_t access_flags, struct ibv_mr **mr_p)
+                           uint64_t access_flags, struct ibv_mr **mr_p,
+                           int silent)
 {
     struct ibv_mr *mr;
+    ucs_log_level_t level = silent ? UCS_LOG_LEVEL_DEBUG : UCS_LOG_LEVEL_ERROR;
```

Comment:
maybe pass 'int silent' to uct_ib_md_print_mem_reg_err_msg instead of level

Reply:
ok, done